modules/ozoneBidAdapter.js

```diff
@@ -274,4 +338,44 @@ export function getTestQuerystringValue() {
   return 0;
 }
 
+// listeners:
+// http://prebid.org/dev-docs/publisher-api-reference.html
+// this is called immediately after the standard targeting has been set.
+// We will now try to add custom targeting data - auction ID and all losing bidders.
+// we will use this instead of having to use in the html page: pbjs.bidderSettings = { ozone: { ... } } - see http://prebid.org/dev-docs/publisher-api-reference.html
+pbjs.onEvent('setTargeting', function(arrData) {
+  utils.logInfo('In the setTargeting event handler. Adding custom k/v to the ad request. _ozoneInternal = ', _ozoneInternal);
+  /*
+  In order to add a custom key/value pair you fundamentally need to do one thing:
+  Set new targeting for this key by calling setTargeting() on each window.googletag.pubads().getSlots() array.
+  */
+  // iterate over the slot objects, adding custom targeting parameters
+  utils.logInfo('going to iterate slots: ', window.googletag.pubads().getSlots());
+  window.googletag.pubads().getSlots().forEach(function (slot) {
```

Comment:
adapters should not be modifying the DFP slots directly. Please drop this part of the code.